internal/commands/test_test.go

```diff
@@ -167,3 +170,50 @@ ENTRYPOINT ["java","-cp","app:app/lib/*","hello.Application"]`
 		t.Errorf("Dockerfile test failure. Got %v successes, expected %v", actualSuccesses, expectedSuccesses)
 	}
 }
+
+func TestGetFilesFromDirectory(t *testing.T) {
+	os.Mkdir("test/", 0700)
+	defer os.RemoveAll("test/")
+
+	createDummyFile := func(name string) {
+		d := []byte("")
+		check(ioutil.WriteFile(name, d, 0644))
+	}
+	err := os.MkdirAll("test/parent/child", 0755)
+	check(err)
+
+	createDummyFile("test/file1.tf")
+	createDummyFile("test/file2.tf")
+	createDummyFile("test/parent/file1.tf")
+	createDummyFile("test/parent/file1.yaml")
+	createDummyFile("test/parent/child/test.tf")
+
+	tests := []struct {
+		regex string
+		exp   []string
+	}{
+		{".*.yaml", []string{"test/file1.tf", "test/file2.tf", "test/parent/child/test.tf", "test/parent/file1.tf"}},
+		{".*.tf", []string{"test/parent/file1.yaml"}},
+		{"child/", []string{"test/file1.tf", "test/file2.tf", "test/parent/file1.tf", "test/parent/file1.yaml"}},
+		{"parent/", []string{"test/file1.tf", "test/file2.tf"}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.regex, func(t *testing.T) {
+			result, err := getFilesFromDirectory("test/", tt.regex)
+			if err != nil {
+				t.Errorf("getFilesFromDirectory returns err, expected nil")
```

Comment:
opinion: This might be slightly preference, but I tend to advocate for t.Fatal in the event that any sort of setup within the test fails. Error will continue the test suite, but there's no point in checking the next case if we didn't manage to get a successful result.

Reply:
I don't mind that, just changed it with t.fatalf